Zend/zend_constants.c

```diff
@@ -100,7 +100,7 @@ void clean_module_constants(int module_number)
 
 void zend_startup_constants(void)
 {
-	EG(zend_constants) = (HashTable *) malloc(sizeof(HashTable));
+	EG(zend_constants) = (HashTable *) emalloc(sizeof(HashTable));
 	zend_hash_init(EG(zend_constants), 128, NULL, ZEND_CONSTANT_DTOR, 1);
```

Comment:
These allocations must be persistent because the lifetime of `EG(zend_constants)` spans multiple requests. The Zend allocator is reset after every request.

See https://www.phpinternalsbook.com/php7/extensions_design/php_lifecycle.html